### PR TITLE
#1098: remove error field from error messages

### LINKED ISSUE #1098: Remove "error" field from errors.
```go
body := &errorBody{
		Error:   s.Message(),
		Message: s.Message(),
		Code:    int32(s.Code()),
		Details: s.Proto().GetDetails(),
}
```

Here ([link](https://github.com/grpc-ecosystem/grpc-gateway/blob/3c06998610d4d2edea99d12228b1f6ea2d943ce4/runtime/errors.go#L109))

The Error and Message both field got s.Message(), I think it's not clean to duplicate the same thing. I don't know if there's any special intention behind it.

Please let us know. Thanks.

### PR DESCRIPTION
**Pull request by srenatus**
Fixes #1098.

⚠️ Doesn't remove `internal/errors.proto`'s `StreamError`. That can be done next ;)

### CODE REVIEW

File: internal/errors.pb.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
 // 	protoc-gen-go v1.21.0
-// 	protoc        v3.10.1
+// 	protoc        v3.7.0
```
Comment (srenatus) [inline internal/errors.pb.go:4]: I can remove this, or commit it from all the other `.pb.go` files where the bazel run has changed just this line. What do you think?

Comment (johanbrandhorst) [reply internal/errors.pb.go:4]: I think we should be able to remove this

Comment (srenatus) [reply internal/errors.pb.go]: (How) do we control the protoc version used?

Comment (johanbrandhorst) [reply internal/errors.pb.go]: It's in the build environment [Dockerfile](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/.circleci/Dockerfile#L20) at the moment. Clearly this file isn't being regenerated.

File: internal/errors.proto [hunk 2]
```diff
@@ -4,17 +4,6 @@ option go_package = "internal";
...
-	// https://github.com/grpc/grpc/blob/master/src/proto/grpc/status/status.proto
-	// It should be the exact same message as the Error field.
-	int32 code = 2;
-	string message = 3;
-	repeated google.protobuf.Any details = 4;
-}
-
 // StreamError is a response type which is returned when
 // streaming rpc returns an error.
 message StreamError {
```
Comment (johanbrandhorst) [inline internal/errors.proto:9]: Can we delete this too? Not sure we need it to be anything other than a `status` either.

Comment (srenatus) [reply internal/errors.proto:9]: Sure! I'll get rid of it once I've dealt with the fallout of removing `Error`. 👍

**Comment by srenatus** [comment]
Looks like removing the Error message type has implications on the openapi spec:
- it's processed in [protoc-gen-swagger/genswagger/template.go](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/protoc-gen-swagger/genswagger/template.go#L966-L977)
- and put into the specs, like this example https://github.com/grpc-ecosystem/grpc-gateway/blob/v2/examples/internal/clients/echo/api/swagger.yaml#L525-L538

Not a blocker, just something I hadn't had on my radar. I'll see into replacing that with the status message (similar to `protobufAny`).

**Comment by srenatus** [comment]
So far, the [generate failures](https://circleci.com/gh/grpc-ecosystem/grpc-gateway/7412?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) look reasonable. Let's drop `StreamError` next.

**Comment by srenatus** [comment]
I haven't fully understood how the StreamError handling works -- I think I'd rather we got this in, and I (or someone else) pick up removing StreamError next. What do you think?

File: examples/internal/integration/integration_test.go [hunk 3]
```diff
@@ -32,7 +32,7 @@ import (
 )
 
 type errorBody struct {
-	Error   string        `json:"error"`
+	Message string        `json:"message"`
 	Code    int           `json:"code"`
 	Details []interface{} `json:"details"`
 }
```
Comment (johanbrandhorst) [inline examples/internal/integration/integration_test.go:38]: Could this type be a `status.Status`?

File: runtime/errors.go [hunk 4]
```diff
@@ -116,6 +115,7 @@ func DefaultHTTPError(ctx context.Context, mux *ServeMux, marshaler Marshaler, w
 	if !ok {
 		s = status.New(codes.Unknown, err.Error())
 	}
+	pb := s.Proto()
```
Comment (johanbrandhorst) [inline runtime/errors.go:118]: We can replace the code above this like with `status.Convert(err)` (it's equivalent AFAIK).

File: runtime/errors.go [hunk 5]
```diff
@@ -116,6 +115,7 @@ func DefaultHTTPError(ctx context.Context, mux *ServeMux, marshaler Marshaler, w
 	if !ok {
```
Comment (johanbrandhorst) [inline runtime/errors.go:115]: The docstring of this function is no longer accurate.

File: examples/internal/integration/integration_test.go (hunk 3 above)
Comment (srenatus) [reply examples/internal/integration/integration_test.go:38]: I believe we've got a test case doing that already. I'll double-check and either remove this one or change it.

Comment (srenatus) [reply examples/internal/integration/integration_test.go:38]: 👉 https://github.com/grpc-ecosystem/grpc-gateway/blob/e6d05361fee32e0f75b57de7b98b9455742384a3/examples/internal/integration/proto_error_test.go#L71 

Looks like we can remove this one, then. Unless I'm mistaking their intention.

**Comment by srenatus** [comment]
@johanbrandhorst I've done a quick grep, but I'm afraid I wasn't very thorough.

**Comment by johanbrandhorst** [comment]
Great work Stephan! Lets get this in :). You've reminded me that we need to have a general look at the error functions, I believe we have too many ways to configure error handling at the moment.

**Comment by ahrtr** [comment]
This PR is a breaking change, could you please add the error field back? Please refer to https://github.com/etcd-io/etcd/issues/21100#issuecomment-3760942241

Thanks

**Pull request merged**